srai/loaders/base.py

```diff
@@ -0,0 +1,22 @@
+"""Base class for loaders."""
+
+import abc
+from typing import Any
+
+import geopandas as gpd
+
+
+class BaseLoader(abc.ABC):
+    """Base abstract class for loaders."""
+
+    @abc.abstractmethod
+    def load(self, gdf: gpd.GeoDataFrame, **kwargs: Any) -> gpd.GeoDataFrame:
```

Comment:
I'm not sure if we should require a GeoDataFrame as an input. I can always try to provide a .geojson/.pbf/<any other> file and ask it to load it as is without filtering.
This function can be used independently by users just to load their data into a data frame using a unified API.